db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -3639,4 +3620,91 @@ public double getDefaultSelectivityFactor() {
     public String getClientIPAddress() {
         return ipAddress;
     }
+
+    @Override
+    public void logStartCompiling(String statement) {
+        stmtLogger.info(String.format("Begin compiling prepared statement. %s, %s",
+                getLogHeader(), formatLogStmt(statement)));
+    }
+
+    @Override
+    public void logEndCompiling(String statement, long nanoTimeSpent) {
+        stmtLogger.info(String.format("End compiling prepared statement. %s, timeSpent=%dms, %s",
+                getLogHeader(), nanoTimeSpent / 1000000, formatLogStmt(statement)));
+    }
+
+    @Override
+    public void logErrorCompiling(String statement, Throwable t, long nanoTimeSpent) {
+        stmtLogger.warn(String.format("Error compiling prepared statement. %s, timeSpent=%dms, %s",
+                getLogHeader(), nanoTimeSpent / 1000000, formatLogStmt(statement)), t);
+    }
+
+    @Override
+    public void logCommit() {
+        stmtLogger.info(String.format("Committing. %s", getLogHeader()));
+    }
+
+    @Override
+    public void logRollback() {
+        stmtLogger.info(String.format("Rolling back. %s", getLogHeader()));
```

Comment:
Please guard all logging with stmtLogger.isInfoEnabled() checks to avoid the formatting cost when logging is disabled.

Reply:
Thanks for the suggestion. I'll fix them in another PR.